enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+with a few workarounds:
+    
+- We need to clone the `etc-pki-entitlement` certificates into the MCO
+  namespace.
+- These need to be mounted into the build pod.
+- These need to be copied from the secret volume mount into a place where
+  Buildah can consume them.
+
+In the future, we could make use of the [shared resource CSI driver](https://docs.openshift.com/container-platform/4.17/storage/container_storage_interface/ephemeral-storage-shared-resource-csi-driver-operator.html)
+to automatically keep this secret up-to-date. In lieu of that, the cluster
+admin will need to keep the secret up-to-date or the MCO will have to take
+on that role as well.
+    
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think
+broadly. For example, consider both security and how this will impact
+the larger OKD ecosystem.
+
+How will security be reviewed and by whom?
+
+- For the custom pod builder, a review by someone who is knowledgeable
+  about container builders, RBAC, and SELinux would be helpful to ensure
+  that we can enable this behavior while avoiding the need for elevated
+  permissions / privileges.
+
+How will UX be reviewed and by whom?
+
+- TBD.
+
+### Drawbacks
+
+TBD.
+
+## Design Details
+
+### Open Questions
+
+- Since bootc can ultimately use ConfigMaps, does it make sense to
+  expose that functionality directly to cluster administrators by
+  allowing them to create / modify / delete ConfigMaps in the MCO
+  namespace to update their node configs? We could use a specific label
+  or annotation to signal to the MCO that it should consider these
+  ConfigMaps as it prepares the rendered MachineConfig.
+
+- What annotations will bootc support for ConfigMaps?
+
+- Will bootc be able to inject ConfigMap contents anywhere on a nodes’
+  filesystem, including a users’ home directory?
+
+- Will bootc ConfigMap changes require a reboot to take effect?
+  Specifically, when bootc is commanded to update contents on the
+  filesystem, will a reboot be required for the content update to occur?
+
+- Will bootc eventually support other CRDs such as MachineConfigs? Or will bootc
+  only support ConfigMaps for the foreseeable future?
+
+- How immutable will RHCOS end up being? Do we wish to enable more
+  immutability? Should a cluster administrator make that determination?
+  Can that adjustment be a Day 2 operation? Would it be desirable to
+  have the RHIVOS-like feature of using overlayfs so that any changes
+  disappear after a reboot? (see
+  [<u>demo</u>](https://drive.google.com/file/d/1RwjxidqnZs6sraEgHjWge6-2msoll1BK/view)
+  for an example).
+
+- How will the relationship with the `osImageURL` field on MachineConfigs
+  change with this new paradigm? Could a cluster admin theoretically use
+  the `.spec.customContent.baseImagePullspec` field to achieve the same
+  functionality? Could the `osImageURL` field be removed from
+  MachineConfigs?
```

Comment:
This would require migrating all existing users away from the field, what might be involved in that?